src/rgw/rgw_op.cc

```diff
@@ -4811,6 +4826,26 @@ void RGWDeleteObj::execute()
       return;
     }
 
+    // cache the objects tags and metadata into the requests
+    // so it could be used in the notification mechanism
+    try {
+      populate_tags_in_request(s, attrs);
+    } catch (buffer::error& err) {
+      ldpp_dout(this, 5) << "WARNING: failed to populate delete request with object tags: " << err.what() << dendl;
+    }
+    populate_metadata_in_request(s, attrs);
+
+    // make reservation for notification if needed
+    // TODO: delete marker is set only after object is deleted
```

Comment:
what happened with this TODO?

Reply:
not needed. event type is checked according to whether versioning is enabled in the bucket, and whether the object being deleted does not have a version. we don't check for the delete marker itself.